database/src/lib.rs

```diff
@@ -72,6 +72,159 @@
 // ).await?,
 
 use scylla::prepared_statement::PreparedStatement;
+use scylla::retry_policy::{QueryInfo, RetryDecision};
+use scylla::transport::errors::QueryError;
+
+#[derive(Debug)]
+pub struct CustomRetryPolicy {
+    max_retry: u8,
+    strict_mode: bool,
+}
+
+impl CustomRetryPolicy {
+    pub fn new(max_retry: u8, strict_mode: bool) -> CustomRetryPolicy {
+        CustomRetryPolicy {
+            max_retry,
+            strict_mode,
+        }
+    }
+}
+
+impl Default for CustomRetryPolicy {
+    /// By default using 5 retries
+    fn default() -> CustomRetryPolicy {
+        CustomRetryPolicy::new(5, false)
+    }
+}
+
+impl scylla::retry_policy::RetryPolicy for CustomRetryPolicy {
+    /// Called for each new query, starts a session of deciding about retries
+    fn new_session(&self) -> Box<dyn scylla::retry_policy::RetrySession> {
+        Box::new(CustomRetrySession::new(self.max_retry, self.strict_mode))
+    }
+
+    /// Used to clone this RetryPolicy
+    fn clone_boxed(&self) -> Box<dyn scylla::retry_policy::RetryPolicy> {
+        Box::new(CustomRetryPolicy::new(self.max_retry, self.strict_mode))
+    }
+}
+
+struct CustomRetrySession {
+    max_retry: u8,
+    max_retry_count: u8,
+    strict_mode: bool,
+}
+
+impl CustomRetrySession {
+    pub fn new(max_retry: u8, strict_mode: bool) -> CustomRetrySession {
+        CustomRetrySession {
+            max_retry,
+            max_retry_count: 0,
+            strict_mode,
+        }
+    }
+
+    pub fn is_retry(&mut self) -> bool {
+        // If `strict_mode` always retry
+        if self.strict_mode {
+            return true;
+        };
+        // If `max_retry_count` is more than `max_retry` don't retry
+        if self.max_retry_count > self.max_retry {
+            return false;
+        };
+        // If `max_retry_count` is less than `max_retry` retry and increment `max_retry_count`
+        self.max_retry_count += 1;
+        true
+    }
+}
+
+impl Default for CustomRetrySession {
+    /// By default using 5 retries
+    fn default() -> CustomRetrySession {
+        CustomRetrySession::new(5, false)
+    }
+}
+
+impl scylla::retry_policy::RetrySession for CustomRetrySession {
+    /// Called after the query failed - decide what to do next
+    fn decide_should_retry(&mut self, query_info: QueryInfo) -> RetryDecision {
+        if let scylla::frame::types::LegacyConsistency::Serial(_) = query_info.consistency {
+            return RetryDecision::DontRetry;
+        };
+        tracing::warn!("Scylla RetryPolicy error: {:?}", query_info.error);
```

Comment:
The message clearly says that the error in RetryPolicy has occurred. But it is misleading. The error occurred in the query.

I suggest changing the message to clearly say about Database error.

Also, I see a huge `match` below dealing with different scenarios, perhaps it's worth adding logs there mentioning the info from the comments those match arms have for our own sake.